### PR TITLE
Bugfix/117 fix phone number validation

### LINKED ISSUE #49: Fix phone number verification
Can't give 077 prefix (HOT)

### LINKED ISSUE #117: Add & fix phone number validation to VolunteerForm
* Currently only the BaseHelpForm has phone number validation.
This validation should occur in VolunteerForm as well.
* Also, this sort of validation does not seem like it would capture all phone numbers. We need to rethink this - maybe it should be a freeform combination of numbers, plus and minus signs limited to a certain amount of characters (see #49)

Don't do before #45 is merged as the phone number validation should be similar to how the ID number validation was done in said issue.

### PR DESCRIPTION
**Pull request by ykrinsky**
Resolves #117 , and fixes #49 .

### CODE REVIEW

File: client/validators.py [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+from django.core.validators import RegexValidator
 
 ID_NUMBER_REGEX = r'^\d{8,9}$'
 
 
+# Checks if the phone number starts with +972/0, and then if the second part is one of:
+# 1. a land line geographic based number - 2/3/4/8/9 and afterwards 7 digits.
+# 2. a new number country prefix - 7 and afterwards 8 digits.
+# 3. a mobile number - 5 and afterwards 8 digits.
+PHONE_NUMBER_REGEX = re.compile(r"^\+?(972|0)(-)?0?(([23489]\d{7})|[5|7]\d{8})$")
```
Comment (stavlocker) [inline client/validators.py:12]: Could be written as: `^(\+972-?|0)(([23489]-?\d{7})|[5|7]\d-?\d{4}-?\d{3})$`

I changed the beginning of the regex as it seemed too complicated to check for +972-? or 0, and added hyphen support.

Comment (astar26) [reply client/validators.py]: Hyphens should also be optional.

File: client/validators.py [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+from django.core.validators import RegexValidator
 
 ID_NUMBER_REGEX = r'^\d{8,9}$'
 
 
+# Checks if the phone number starts with +972/0, and then if the second part is one of:
+# 1. a land line geographic based number - 2/3/4/8/9 and afterwards 7 digits.
+# 2. a new number country prefix - 7 and afterwards 8 digits.
+# 3. a mobile number - 5 and afterwards 8 digits.
+PHONE_NUMBER_REGEX = re.compile(r"^(\+972[- ]?|0)(([23489]|[57]\d)-?\d{3}-?\d{4})$")
```
Comment (stavlocker) [inline client/validators.py:12]: It's kind of nit-picky but maybe switch to
```regex
^(\+972-?|0)(([23489]-?\d{7})|[5|7]\d-?(\d{4}-?\d{3}|\d{3}-?\d{4}))$
```
so 054-1234-567 and 054-123-4567 will both be allowed.

**Comment by stavlocker** [comment]
Also, could you fix the two linting errors @github-actions raised?

File: client/validators.py (hunk 1 above)
Comment (ykrinsky) [reply client/validators.py]: Added hyphens support as optional. Also added the option to have whitespace after +972.

File: client/validators.py (hunk 2 above)
Comment (astar26) [reply client/validators.py:12]: Why don't you simply remove hyphens altogether? would greatly simplify the regex and reduce possible errors.

Comment (astar26) [reply client/validators.py:12]: Also would standarize numbers.

Comment (ykrinsky) [reply client/validators.py:12]: Ok, this is the standard format 054-123-4567 but I don't want to confuse the users, so both ways will be allowed.
Also, I will add the option to have whitespace after 050

Comment (stavlocker) [reply client/validators.py]: @astar26 the hyphens are optional.
@ykrinsky I talked to @astar26 , you can just use the solution I gave above for the hyphens and that's alright.

**Pull request merged**